plugin/pkg/admission/nodeenv/admission.go

```diff
@@ -0,0 +1,187 @@
+package nodeenv
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/yaml"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+var NamespaceNodeSelectors = []string{"kubernetes.io/node-selector"}
+
+func init() {
+	admission.RegisterPlugin("PodNodeEnvironment", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewPodNodeEnvironment(client, readConfig(config))
+	})
+}
+
+// podNodeEnvironment is an implementation of admission.Interface.
+type podNodeEnvironment struct {
+	*admission.Handler
+	client clientset.Interface
+	store  cache.Store
+	// global default node selector in a cluster, If a namespace is
+	// not assigned any node selector, it gets this by default.
+	clusterDefaultNodeSelector string
+}
+
+type pluginConfig struct {
+	PodNodeEnvironmentPluginConfig map[string]string
```

Comment:
> separate apiVersion/kind just for configuring default cluster value for a plugin?

Unfortunately, since kube works on a single shared controller config file, you have no reasonable way to manage that.